src/scripts/main.js

```diff
@@ -1,3 +1,430 @@
+
+      case 'Age':
+        return a - b;
+
+      case 'Salary':
+        return normalizeNumber(a) - normalizeNumber(b);
+    }
+  });
+
+  if (dscSorting) {
+    rows.reverse();
+  }
+
+  tbody.append(...rows);
+
+  function normalizeNumber(string) {
+    return +string.slice(1).split(',').join('');
+  }
+}
+
+// show active row
+
+tbody.addEventListener('click', eventClick => {
+  const tr = eventClick.target.closest('tr');
+
+  rows.forEach(row => {
+    row.removeAttribute('class');
+  });
+
+  tr.classList.add('active');
+});
+
+// create form and validate data
+
+const addEmployee = document.createElement('form');
+
+addEmployee.classList.add('new-employee-form');
+
+addEmployee.innerHTML = `
+  <label> Name:
+    <input type="text" name = "name" data-qa="name" required>
+  </label>
+  <label> Position:
+    <input type="text" name = "position" data-qa="position">
+  </label>
+  <label> Office:
+    <select name = "office" data-qa="office" required>
+      <option value="" selected disabled>Choose office</option>
+      <option value="Tokyo">Tokyo</option>
+      <option value="Singapore">Singapore</option>
+      <option value="London">London</option>
+      <option value="New York">New York</option>
+      <option value="Edinburgh">Edinburgh</option>
+      <option value="San Francisco">San Francisco</option>
+    </select>
+  </label>
+  <label> Age:
+    <input type="number" name = "age" data-qa="age" required>
+  </label>
+  <label> Salary:
+    <input type="number" name = "salary" data-qa="salary" required>
+  </label>
+  <button type="submit">Save to table</button>
+`;
+
+body.append(addEmployee);
+
+function validation(data) {
+  if (data.get('name').trim().length < nameLength
+    || /\d/.test(data.get('name'))) {
+    pushNotification(
+      580,
+      10,
+      'Enter a correct name!',
+      'Name must contain at least 4 letters and not include numbers',
+      'error'
+    );
+
+    isValid = false;
+  } else if (data.get('age') < minAge
+  || data.get('age') > maxAge) {
+    pushNotification(
+      580,
+      10,
+      'Enter a correct age!',
+      'Your age must be from 18 to 90 years',
+      'error'
+    );
+
+    isValid = false;
+  } else if (data.get('position').trim().length < nameLength
+    || /\d/.test(data.get('position'))) {
+    pushNotification(
+      580,
+      10,
+      'Enter a position!',
+      'You should enter a correct position',
+      'error'
+    );
+
+    isValid = false;
+  } else if (data.get('salary') <= 0) {
+    pushNotification(
+      580,
+      10,
+      'Enter a correct salary!',
+      'Salary must be more than 0',
+      'error'
+    );
+
+    isValid = false;
+  } else {
+    isValid = true;
+  }
+};
+
+addEmployee.addEventListener('submit', submitEvent => {
+  const employeeInfo = new FormData(addEmployee);
+  const salaryToNormalize = employeeInfo.get('salary');
+  let normalizedSalary = salaryToNormalize;
+
+  if (salaryToNormalize.length > 3) {
```

Comment:
It works only with the first 3 characters:
![image](https://user-images.githubusercontent.com/86923089/219942991-5af07b30-9b03-4cc7-8452-a356f2db8beb.png)
[toLocaleString()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toLocaleString) will be useful here